### ISSUE TITLE
Missing excess property error with spread

### ISSUE THREAD

**Issue by andy-ms** [microsoft/TypeScript#27355]
**TypeScript Version:**  3.1.0-dev.201xxxxx

**Code**

```ts
function f(o: { a: number, b: number }): { a: number } {
    return { b: 0, ...o };
}
```

**Expected behavior:**

Error on `b`.

**Actual behavior:**

No error.

Broken by #26798.

**Related Issues:** #27166, #23340

**Comment by sandersn** [comment]
@weswigham points out that this is probably correct; `b: 0` is not excess because it's overridden by `o.b` which comes from a spread, so should not count as excess. 

What would be more useful is an error for unused properties in a spread: any property that is not used because of later spreads. The error should only apply to non-spread properties.

**Comment by WreckedArrow** [comment]
I'm not sure if this is related or has been addressed elsewhere, but this also does not produce an error:
```typescript
function f(o: { a: number, b: number }): { a: number } {
    return o;
}
```

**Comment by andy-ms** [comment]
@WreckedArrow That's #12936
